File: java/multiply_strings.java
public class Solution {
    public String multiply(String num1, String num2) {
        int m=num1.length();
        int n=num2.length();
        if (m==0||n==0) return "";
        StringBuilder sb=new StringBuilder();
        int carry=0, num;
        
        for (int j=n-1; j>=0; j--) {
            for (int i=m-1; i>=0; i--) {
                num=(num1.charAt(i)-'0')*(num2.charAt(j)-'0')+carry;
                carry=num/10;
                num%=10;
                int index=(n-1-j)+(m-1-i);
                if (index<sb.length()) {
                    int sum=num+sb.charAt(index)-'0';
                    if (sum>9) {
                        carry++;
                        sum%=10;
                    }
                    sb.setCharAt(index, (char)(sum+'0'));
                } else {
                    sb.append((char)(num+'0'));
                }
            }
            if (carry>0) sb.append((char)(carry+'0'));
            carry=0;
        }

        int k;
        for (k=sb.length()-1; k>0; k--) {
            if (sb.charAt(k)!='0') break;
        }   
        sb.delete(k+1, sb.length());
        
        sb.reverse();
        return sb.toString();
    }
}

File: python/multiply-strings.py
class Solution:
    def multiply(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        nums=[0]*(len(num1)+len(num2))
        for i in reversed(range(len(num1))):
            for j in reversed(range(len(num2))):
                val=(ord(num1[i])-ord('0'))*(ord(num2[j])-ord('0'))
                p1, p2=i+j, i+j+1
                val+=nums[p2]
                nums[p1]+=int(val/10)
                nums[p2]=val%10
                
        k=0
        while k<len(nums)-1 and nums[k]==0: k+=1
        return ''.join(map(str, nums[k:]))
